python/lsst/obs/fiberspectrograph/translator.py

```diff
@@ -0,0 +1,214 @@
+import logging
+import os
+
+import astropy.units as u
+from astropy.time import Time
+
+from astro_metadata_translator import cache_translation
+from lsst.obs.lsst.translators.lsst import SIMONYI_TELESCOPE, LsstBaseTranslator
+
+from lsst.utils import getPackageDir
+
+__all__ = ["FiberSpectrographTranslator", ]
+
+log = logging.getLogger(__name__)
+
+
+class FiberSpectrographTranslator(LsstBaseTranslator):
+    """Metadata translator for Rubin calibration fibre spectrographs headers"""
+
+    name = "FiberSpectrograph"
+    """Name of this translation class"""
+
+    supported_instrument = "FiberSpec"
+    """Supports the Rubin calibration fibre spectrographs."""
+
+    default_search_path = os.path.join(getPackageDir("obs_fiberspectrograph"), "corrections")
+    """Default search path to use to locate header correction files."""
+
+    default_resource_root = os.path.join(getPackageDir("obs_fiberspectrograph"), "corrections")
+    """Default resource path root to use to locate header correction files."""
+
+    DETECTOR_MAX = 1
+
+    _const_map = {
+        "detector_num": 0,
+        "detector_name": "0",
+        "exposure_group": None,
+        "object": None,
+        "physical_filter": "empty",
+        "detector_serial": "0xdeadbeef",
+        "detector_group": "None",
+        "relative_humidity": None,
+        "pressure": None,
+        "temperature": None,
+        "focus_z": None,
+        "boresight_airmass": None,
+        "boresight_rotation_angle": None,
+        "tracking_radec": None,
+        "telescope": SIMONYI_TELESCOPE,
+        "observation_type": "spectrum",  # IMGTYPE is ''
+    }
+    """Constant mappings"""
+
+    _trivial_map = {
+        "observation_id": "OBSID",
+        "science_program": ("PROGRAM", dict(default="unknown")),
+    }
+    """One-to-one mappings"""
+
+    @classmethod
+    def can_translate(cls, header, filename=None):
+        """Indicate whether this translation class can translate the
+        supplied header.
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the header is recognized by this class. `False`
+            otherwise.
+        """
+
+        return "INSTRUME" in header and header["INSTRUME"] in ["FiberSpectrograph.Broad"]
+
+    @cache_translation
+    def to_instrument(self):
+        return "FiberSpec"
```

Comment:
This can be a const mapping but if there are different fiber spectrographs that have to be treated differently then this will need special subclasses (and maybe that explains why SERIAL is not being handled).